contracts/DecentHats.sol

```diff
@@ -0,0 +1,485 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {Enum} from "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+import {IAvatar} from "@gnosis.pm/zodiac/contracts/interfaces/IAvatar.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC6551Registry} from "./interfaces/IERC6551Registry.sol";
+import {IHats} from "./interfaces/hats/full/IHats.sol";
+import {LockupLinear, Broker} from "./interfaces/sablier/full/types/DataTypes.sol";
+import {DecentAutonomousAdmin} from "./DecentAutonomousAdmin.sol";
+import {IHatsModuleFactory} from "./interfaces/hats/full/IHatsModuleFactory.sol";
+import {IHatsElectionEligibility} from "./interfaces/hats/full/IHatsElectionEligibility.sol";
+import {ModuleProxyFactory} from "@gnosis.pm/zodiac/contracts/factory/ModuleProxyFactory.sol";
+import {ISablierV2LockupLinear} from "./interfaces/sablier/ISablierV2LockupLinear.sol";
+
+contract DecentHats {
+    string public constant NAME = "DecentHats";
+    bytes32 public constant SALT =
+        0x5d0e6ce4fd951366cc55da93f6e79d8b81483109d79676a04bcc2bed6a4b5072;
+
+    struct SablierStreamParams {
+        ISablierV2LockupLinear sablier;
+        address sender;
+        address asset;
+        LockupLinear.Timestamps timestamps;
+        Broker broker;
+        uint128 totalAmount;
+        bool cancelable;
+        bool transferable;
+    }
+
+    struct TermedParams {
+        uint128 termEndDateTs;
+        address[] nominatedWearers;
+    }
+
+    struct Hat {
+        address wearer;
+        string details;
+        string imageURI;
+        SablierStreamParams[] sablierParams;
+        TermedParams[] termedParams;
+        uint32 maxSupply;
+        bool isMutable;
+        bool isTermed;
```

Comment:
Do we need to have this boolean? Can we not infer it based on the existence of any items in the `termedParams` array?

Reply:
fixed, thanks